src/sagemaker/amazon/amazon_estimator.py

```diff
@@ -152,9 +153,98 @@ def record_set(self, train, labels=None, channel="train"):
         return RecordSet(manifest_s3_file, num_records=train.shape[0], feature_dim=train.shape[1], channel=channel)
 
 
+class AmazonS3AlgorithmEstimatorBase(EstimatorBase):
+    """Base class for Amazon first-party Estimator implementations. This class isn't
+    intended to be instantiated directly. This is difference from the base class
+    because this class handles S3 data"""
+
+    mini_batch_size = hp('mini_batch_size', (validation.isint, validation.gt(0)))
```

Comment:
please look at https://github.com/aws/sagemaker-python-sdk/pull/54 we intentionally removed these type validations: isint() isbool() etc. In favor of declaring a specific type for the hp.

So this should be

`hp('mini_batch_size', validation.gt(0), data_type=int)`

This applies to every hp declaration in this PR.